internal/datagen/generator.go

```diff
@@ -0,0 +1,77 @@
+package datagen
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"math/rand"
+	"time"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/modules"
+)
+
+type (
+	// Generator stores buffer of random bytes with some tail and returns data slices with
+	// an increasing offset so that we receive a different set of bytes after each call without
+	// re-generation of the entire buffer from scratch:
+	//
+	//   [<----------size----------><-tail->]
+	//   [<----------slice0-------->........]
+	//   [.<----------slice1-------->.......]
+	//   [..<----------slice2-------->......]
+	Generator struct {
+		vu     modules.VU
+		size   int
+		buf    []byte
+		offset int
+	}
+
+	GenPayloadResponse struct {
+		Payload goja.ArrayBuffer
+		Hash    string
+	}
+)
+
+// TailSize specifies number of extra random bytes in the buffer tail.
+const TailSize = 1024
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+}
+
+func NewGenerator(vu modules.VU, size int) Generator {
```

Comment:
Let's write some unit tests for the generator, since it has zero dependencies with other components unlike native or s3 client.

Use [moduletest](https://github.com/grafana/k6/blob/63054efc4bc185b4961ace8a4b6df81d220316d8/js/modulestest/modulestest.go#L15) package to mock VU. Something like this (spoiler: we don't process negative sizes).

```go
import (
	"testing"

	"github.com/dop251/goja"
	"github.com/stretchr/testify/require"
	"go.k6.io/k6/js/modulestest"
)

func TestGenerator(t *testing.T) {
	vu := &modulestest.VU{
		RuntimeField: goja.New(),
	}

	t.Run("negative size", func(t *testing.T) {
		require.Panics(t, func() {
			_ = NewGenerator(vu, -1)
		})
	})
}
```